src/query/service/src/sql/optimizer/rule/rewrite/rule_push_down_limit_sort.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp;
+
+use crate::sql::optimizer::rule::Rule;
+use crate::sql::optimizer::rule::TransformState;
+use crate::sql::optimizer::RuleID;
+use crate::sql::optimizer::SExpr;
+use crate::sql::plans::Limit;
+use crate::sql::plans::PatternPlan;
+use crate::sql::plans::RelOp;
+use crate::sql::plans::RelOp::Pattern;
+use crate::sql::plans::RelOp::Sort;
+use crate::sql::plans::RelOperator;
+use crate::sql::plans::Sort as logsort;
+
+/// Input:  Limit
+///           \
+///          Sort
+///             \
+///              *
+///
+/// Output: Limit
+///           \
+///          Sort
+///             \
+///            Limit
```

Comment:
We can't add `limit` before the sort!

Reply:
Limit --> sort --> Get

-->

Limit -> Sort --- > Get {Limit, Sort }

Reply:
Modify the comment.